arcgishub/sites.py

```diff
@@ -413,11 +436,96 @@ def update(self, site_properties=None, data=None, thumbnail=None, metadata=None)
+                _site_domain = requests.get(path, headers = _HEADERS)
+                _siteId = _site_domain.json()[0]['id']
+                client_key = _site_domain.json()[0]['clientKey']
+                #Delete old domain entry
+                _HEADERS = {
+                    'Content-Type': 'application/json', 
+                    'Authorization': self._gis._con.token, 
+                    'Referer': self._gis._con._referer
+                }
+                path = 'https://hub.arcgis.com/utilities/domains/'+_siteId
+                _delete_domain = requests.delete(path, headers = _HEADERS)
+                #if deletion is successful
+                if _delete_domain.status_code==200:
+                    #Create new domain entry
+
+                    #Create domain entry for new site
+                    _HEADERS = {
+                        'Content-Type': 'application/json', 
+                        'Authorization': self._gis._con.token,
+                        'Referer': self._gis._con._referer
+                    }
+                    _body = {
+                        'hostname': subdomain + '-' + self._gis.properties['urlKey'] + '.hub.arcgis.com', 
+                        'siteId': self.item.id, 
+                        'siteTitle': self.title, 
+                        'clientKey': client_key, 
+                        'orgId': self._gis.properties.id, 
+                        'orgKey': self._gis.properties['urlKey'], 
+                        'orgTitle':self._gis.properties['name'],
+                        'sslOnly':True
+                    }
+                    path = 'https://hub.arcgis.com/utilities/domains'
+                    _new_domain = requests.post(path, headers = _HEADERS, data=json.dumps(_body))
+                    if _new_domain.status_code==200:
+                        #define new domain and hostname
+                        domain = self._gis.url[:8] + subdomain + '-' + self._gis.properties['urlKey'] + '.hub.arcgis.com'
+                        hostname = subdomain + '-' + self._gis.properties['urlKey'] + '.hub.arcgis.com'
+                        #update initiative item 
+                        self.initiative.item.update(item_properties={'url':domain})
+                        #update site item and data
+                        data = self.definition
+                        data['values']['defaultHostname'] = hostname
+                        data['values']['subdomain'] = subdomain
+                        data['values']['internalUrl'] = hostname
+                        if self.item.update(item_properties={'url':domain, 'text':data}):
+                            return domain
+                    #if creating new domain entry fails
+                    else:
+                        return _new_domain.content
+                #if deleting old domain entry fails
+                else:
+                    return _delete_domain.content
+            #For enterprise sites
+            else:
```

Comment:
generally there is a lot of repeat logic in Enterprise & Online, consider how to consolidate that to internal functions